### PR TITLE
Add common data type support

### PR DESCRIPTION
**Pull request by MateuszKubuszok**
Provides #9.

### CODE REVIEW

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala [hunk 1]
```diff
@@ -49,4 +53,54 @@ object DerivedTransformer {
...
+  : DerivedTransformer[Option[From], Option[To], Modifiers] =
+    (src: Option[From], modifiers: Modifiers) => src.map(innerTransformer.transform(_, modifiers))
+
+  implicit def someTransformer[From, To, Modifiers <: HList]
+    (implicit innerTransformer: DerivedTransformer[From, To, Modifiers])
+  : DerivedTransformer[Some[From], Option[To], Modifiers] =
+    (src: Some[From], modifiers: Modifiers) => Some(innerTransformer.transform(src.value, modifiers))
+
+  implicit def leftTransformer[FromL, ToL, FromR, ToR, Modifiers <: HList]
+  (implicit leftTransformer: DerivedTransformer[FromL, ToL, Modifiers])
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: please align indentation

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala [hunk 2]
```diff
@@ -49,4 +53,54 @@ object DerivedTransformer {
...
+  : DerivedTransformer[Left[FromL, FromR], Left[ToL, ToR], Modifiers] =
+    (src: Left[FromL, FromR], modifiers: Modifiers) => Left(leftTransformer.transform(src.value, modifiers))
+
+  implicit def rightTransformer[FromL, ToL, FromR, ToR, Modifiers <: HList]
+  (implicit rightTransformer: DerivedTransformer[FromR, ToR, Modifiers])
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: indentation

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala [hunk 3]
```diff
@@ -49,4 +53,54 @@ object DerivedTransformer {
     (src: From, modifiers: Modifiers) =>
       toLG.from(genTransformer.transform(tag[From](fromLG.to(src)), modifiers))
   }
+
+  implicit def optionTransformer[From, To, Modifiers <: HList]
+    (implicit innerTransformer: DerivedTransformer[From, To, Modifiers])
+  : DerivedTransformer[Option[From], Option[To], Modifiers] =
+    (src: Option[From], modifiers: Modifiers) => src.map(innerTransformer.transform(_, modifiers))
+
+  implicit def someTransformer[From, To, Modifiers <: HList]
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: I'm curious if single instance can be sufficient (for `X <: Option[T]`). Similarly for Either.

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala (hunk 1 above)
Comment (MateuszKubuszok) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: :+1:

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala (hunk 2 above)
Comment (MateuszKubuszok) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: :+1:

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala (hunk 3 above)
Comment (MateuszKubuszok) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: :+1:

File: chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala [hunk 4]
```diff
@@ -118,6 +118,45 @@ class DslSpec extends WordSpec with MustMatchers {
...
+    "support common data types" should {
+
+      case class Foo(value: String)
+      case class Bar(value: String)
+
+      "support scala.Option" in {
+        Option(Foo("a")).transformInto[Option[Bar]] mustBe Option(Bar("a"))
+        (Some(Foo("a")): Option[Foo]).transformInto[Option[Bar]] mustBe Option(Bar("a"))
+        Some(Foo("a")).transformInto[Option[Bar]] mustBe Some(Bar("a")) // FIXIT
+        None.transformInto[Option[Bar]] mustBe None                     // FIXIT
```
Comment (MateuszKubuszok) [inline chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala]: We can remove `// FIXIT` now ;)

**Pull request merged**